liquidprompt

```diff
@@ -1397,9 +1424,53 @@ _lp_kubernetes_context() {
 _lp_kubernetes_context_color() {
     _lp_kubernetes_context || return "$?"
 
-    lp_kubernetes_context_color="[${LP_MARK_KUBECONTEXT}${LP_COLOR_KUBECONTEXT}${lp_kubernetes_context}${lp_kubernetes_namespace+:}${lp_kubernetes_namespace-}${NO_COL}]"
+    lp_kubernetes_context_color="${LP_MARK_KUBECONTEXT}${LP_COLOR_KUBECONTEXT}${lp_kubernetes_context}${lp_kubernetes_namespace+:}${lp_kubernetes_namespace-}${NO_COL}"
 }
 
+_lp_cmake() {
+    (( LP_ENABLE_CMAKE )) || return 2
+    [[ ! -f CMakeCache.txt ]] && return 1
+
+    local cmake_compiler
+    IFS="=" read -r _ cmake_compiler <<<"$(grep COMPILER:FILEPATH CMakeCache.txt)"
+    lp_cmake_compiler="${cmake_compiler##*/}" # Only the part after the last slash.
+
+    local cmake_generator
+    IFS="=" read -r _ cmake_generator <<<"$(grep CMAKE_GENERATOR:INTERNAL CMakeCache.txt)"
+    # Shorten: Makefiles -> Make, and Visual Studio -> VS.
+    cmake_generator="${cmake_generator/%Makefiles/Make}"
+    lp_cmake_generator="${cmake_generator/#Visual Studio/VS}"
+
+    local cmake_buildtype
+    IFS="=" read -r _ lp_cmake_buildtype <<<"$(grep CMAKE_BUILD_TYPE:STRING CMakeCache.txt)"
+
+    [[ -z "${lp_cmake_compiler}${lp_cmake_generator}${lp_cmake_buildtype}" ]] && return 3
+
+    lp_cmake="${lp_cmake_compiler}${LP_MARK_CMAKE}${lp_cmake_generator}${LP_MARK_CMAKE}${lp_cmake_buildtype}"
```

Comment:
> SC2034 (warning): lp_cmake appears unused. Verify use (or export if used externally).

Yeah, it is unused. I think you should just delete this line altogether, and just return your 3 data strings and expect the theme to figure them out.